src/riscv-integration.adoc

```diff
@@ -1078,27 +1078,41 @@ memory addresses must match for the address to be valid:
 * For Sv48, bits [63:48] must equal bit 47
 * For Sv57, bits [63:57] must equal bit 56
 
-RISC-V permits that some CSRs, such as <<mtvec>> and <<mepc>> (see
-xref:CSR_exevectors[xrefstyle=short]), need not be able to hold all possible
-invalid addresses. Prior to writing these CSRs, implementations may convert an
+RISC-V permits that some CSRs holding addresses, such as <<mtvec>> and <<mepc>>
+(see xref:CSR_exevectors[xrefstyle=short]) and consequently <<pcc>>, need not be able to hold all possible
+invalid addresses. Prior to writing these registers, implementations may convert an
 invalid address into some other invalid address that the register is capable of
 holding. However, these registers hold capabilities in {cheri_base_ext_name}
 and the bounds encoding depends on the address value, so implementations must
 not convert invalid addresses to other arbitrary invalid addresses in an
 unrestricted manner.
 The following procedure must be used instead when writing a capability A to
-these CSRs:
+a CSR:
 
-. If A's address cannot be held then convert it to another address that the CSR can
-hold
-. If conversion _was_ required, then A's tag is cleared if A is
-sealed or if the new address is not representable -- this is equivalent to the
-semantics of <<SCADDR>>
+. If A's address cannot be held, then convert it to another address that the
+register can hold.
+. If A's address is invalid and A's bounds do not equal the bounds of the
```

Comment:
Infinite capability should be an xref, in all references in the text.

it's worth spelling out how to make the bounds match - either here or in the xref - all we're really saying is that EXP is 0 representing maximum bounds. There's no need to look at the T and B fields, for example (once @PRugg-Cap has merged the 66-bit arithmetic PR)

it's also worth making that a clear statement _first_ that all CSRs or registers which store an address, where the full bit width isn't stored in an RV64 VM core, must always clear the tag if the address is invalid _and_ the bounds aren't infinity.

the rest of the text is fine, but having that statement first will make it clearer.

Reply:
Yeah, I'd be tempted to reword "do not equal the bounds of the Infinite capability" to "does not have infinite bounds", and link to a section somewhere that defines that in text as `base <= 0, top >= 2**xlenmax`, with a note saying that for tagged (and in fact non-malformed) capabilities, this is equivalent to checking EXP is `exp_max` or I guess 0 now as Tariq says.

Reply:
I addressed these comments. Could you please take another look? Thanks!

Reply:
is this ok now @PRugg-Cap ?